src/NotificationInline/index.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import CheckmarkIcon from '../assets/icons/checkmarkCircled'
+import CloseIcon from '../assets/icons/closeCircled'
+import InfoIcon from '../assets/icons/infoCircled'
+import QuestionIcon from '../assets/icons/questionCircled'
+import WarningIcon from '../assets/icons/warningCircled'
+import Spinner from '../Spinner'
+import Text from '../Text'
+import View from '../View'
+
+// =======================================================
+// Notification Types
+// =======================================================
+
+const notificationTypes = {
+   error: {
+      backgroundColor: 'rgba(247, 145, 132, 0.9)',
+      icon: CloseIcon,
+      iconColor: 'rgb(198, 73, 60)',
+      textColor: 'rgb(102, 36, 36)'
+   },
+   info: {
+      backgroundColor: 'rgb(161, 212, 255)',
+      icon: InfoIcon,
+      iconColor: 'rgb(66, 134, 238)',
+      textColor: 'rgb(34, 74, 129)'
+   },
+   loading: {
+      backgroundColor: 'rgb(84, 84, 84)',
+      icon: Spinner,
+      iconColor: 'rgb(134, 134, 134)',
+      textColor: 'rgb(255, 255, 255)'
+   },
+   prompt: {
+      backgroundColor: 'rgb(111, 231, 216)',
+      icon: QuestionIcon,
+      iconColor: 'rgb(11, 181, 181)',
+      textColor: 'rgb(17, 93, 89)'
+   },
+   success: {
+      backgroundColor: 'rgb(134, 240, 180)',
+      icon: CheckmarkIcon,
+      iconColor: 'rgb(45, 193, 100)',
+      textColor: 'rgb(16, 65, 42)'
+   },
+   warning: {
+      backgroundColor: 'rgb(255, 200, 136)',
+      icon: WarningIcon,
+      iconColor: 'rgb(215, 143, 76)',
+      textColor: 'rgb(102, 66, 19)'
+   }
+}
+
+// =======================================================
+// Component
+// =======================================================
+
+const NotificationInline = ({
+   backgroundColor,
+   borderRadius,
+   buttonPlacement,
+   buttons,
+   children,
+   iconColor,
+   iconSpacing,
+   iconSize,
+   id,
+   message,
+   paddingHorizontal,
+   paddingVertical,
+   style,
+   textColor,
+   textSize,
+   title,
+   type
+}) => {
+   const notificationBackgroundColor = backgroundColor || notificationTypes[type].backgroundColor
+   const notificationIconColor = iconColor || notificationTypes[type].iconColor
+   const notificationTextColor = textColor || notificationTypes[type].textColor
+
+   // TODO: In the future, we will allow users to pass their own custom icon
+   const NotificationIcon = notificationTypes[type].icon
+   const notificationButtons = buttons && buttons.length > 0 && (
```

Comment:
No need to change this, but more just as a side note which you may already know, but you could always make `buttons` have a default val of `[]` so you don't have to worry about checking if it exists and its length